### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of null (reading 'before')

### DISCUSSION THREAD

**Discussion by A1exMiranda** [livewire/livewire#7819] [1. Help] [answered]
Hello everyone, I hope you're doing well.
I've recently upgraded from livewire v2 to v3 with some errors. I managed to break down the majority of them like the one with laravel breeze and alpine but I'm still getting an error on a search input I have on my platform. 

![image](https://github.com/livewire/livewire/assets/10999395/73c301aa-6a0c-4deb-970e-8e03466c9a60)

Typing on input

![image](https://github.com/livewire/livewire/assets/10999395/7d137a6d-a069-4cc0-86e9-8d3ce93c3e91)

Console error 

![image](https://github.com/livewire/livewire/assets/10999395/bbaa38b6-8799-4d73-8ebd-87d8132350c7)

```
TypeError: Cannot read properties of undefined (reading 'before')
    at Block.appendChild (livewire.js?id=8ed4c109:8126:23)
    at patchChildren (livewire.js?id=8ed4c109:7954:21)
    at patchChildren (livewire.js?id=8ed4c109:7996:11)
    at patchChildren (livewire.js?id=8ed4c109:7996:11)
    at patch (livewire.js?id=8ed4c109:7887:7)
    at patchChildren (livewire.js?id=8ed4c109:8043:9)
    at patch (livewire.js?id=8ed4c109:7887:7)
    at patchChildren (livewire.js?id=8ed4c109:8043:9)
    at patch (livewire.js?id=8ed4c109:7887:7)
    at patchChildren (livewire.js?id=8ed4c109:8043:9)
```

Removing the previously typed data on input displays only one result

![image](https://github.com/livewire/livewire/assets/10999395/eed3e58e-4c6a-41b0-bb82-c310ee675de2)

```
class InvoicesPagination extends ModalComponent
{
    use WithPagination;
    
    public $search;
    public $sortColumn='name';
    public $sortDirection='asc';
    public int $invoice_id;

    public function sort($column)
    {
        $this->sortColumn = $column;
        $this->sortDirection = $this->sortDirection == 'asc' ? 'desc' : 'asc';
    }

    public function mount()
    {
        $this->search = '';
    }

    public function render(Request $request)
    {
        $request->name = $this->search;

        $invoices = (new InvoiceSearch)->filter($request)->orderBy($this->sortColumn, $this->sortDirection)->paginate(50)->withQueryString();

        return view('livewire.invoices-pagination', ['invoices' => $invoices]);
    }
}

```

```
<div wire:poll.60s>
  <input wire:model.live="search" type="search" id="search" name="search"
                  style="background-color: #f9fafb;"
                  class="block w-full p-4 pl-10 text-sm text-gray-900 border border-gray-300 rounded-lg bg-gray-50 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  placeholder="Buscar Facturas...">
</div>
```

It used to work in v2. 

Thanks in advance!

**Comment by joshhanley** [comment]
@A1exMiranda make sure you have removed AlpineJS from your app.js as it's now bundled with Livewire.

**Comment by DanielGSoftware** [comment]
@A1exMiranda did you manage to find a solution? We have the same problem, and the suggested answer by @joshhanley unfortunately didn't apply to us as we already removed Alpine.

**Reply by A1exMiranda** [reply]
I haven't been able to solve it even though I removed alpine, any suggestion as what can we could do would be appreciated it :)

**Reply by DanielGSoftware** [reply, marked as answer]
I found the solution for my case, I don't know if you have `@continue` in the code where it breaks, but have a look at https://github.com/livewire/livewire/discussions/6965#discussioncomment-8219373

**Reply by A1exMiranda** [reply]
The thread you mentioned had a solution for me, I changed the 'inject_morph_markers' from the livewire.php config to false. Thank you @DanielGSoftware

**Reply by DanielGSoftware** [reply]
@A1exMiranda No worries, glad you found your solution! One thing I do wonder is the potential effect that disabling that config value can have, as it's supposed to deal with other dom diffing issues that were present in V2 I believe. Not to kill the good mood after you found your solution haha, but wanted to tell you this just in case!

**Reply by A1exMiranda** [reply]
Thanks for the info, I'll be looking and testing our aplication to see if another solution comes our way and what are the effects of setting this value to false. Happy coding :)

**Reply by matthewdanepeavoy** [reply]
> I found the solution for my case, I don't know if you have `@continue` in the code where it breaks, but have a look at [#6965 (comment)](https://github.com/livewire/livewire/discussions/6965#discussioncomment-8219373)

The @continue was my issue as well. Still strange error but at least I have a workaround instead now.

**Comment by NBA707** [comment]
I am getting this error as well.  I have a select and then build a table listing child values for the selected item.

I tried inject_morph_markers to false, I don't have a @continue in my @forelse loop.   I assigned a wire:key to each TR tag, put a wire:key in the TABLE tag.

Not sure.   If I change inject_more_markers to false do I need to clear the cache or anything?

**Comment by NBA707** [comment]
I moved my $loop->first and $loop->last statements outside of the foreach and replaced them with count($items) and the error went away.   Interesting.

**Comment by akamasAT** [comment]
had this console error as well - my problem was a very simple one - a missing </tr> in a loop...
code worked fine for months, made a tiny change and all "exploded".

**Reply by davidcamarda** [reply]
Closing </li> tag in a loop broke it for me.

**Comment by JuanDeLeon** [comment]
I had the same issue (migrating to Livewire 3) but within a table, I was doing this:
```
@forelse($collection as $item)
   <tr>
      <td>{{$item->name}}</td>
   </tr>
@empty
   <td>No records</td>
@endforelse
```

Notice the empty block did not have a tr, just by adding the appropriate tr element fixed the issue.

**Comment by leystudio** [comment]
**Livewire 3 + Vite: "Cannot set properties of null (setting '__livewire')" y "livewire is not defined" – ¿Cómo lo solucioné?**

Problema:
Al integrar Laravel Livewire 3 con Vite en mi proyecto, comencé a recibir los siguientes errores en la consola del navegador:

`Uncaught ReferenceError: livewire is not defined
Cannot set properties of null (setting '__livewire')
Uncaught SyntaxError: Invalid regular expression: missing /
`
Al abrir /livewire/livewire.js en el navegador, no se cargaba el script correctamente, y a veces se devolvía HTML en lugar de JavaScript.


_**Contexto:**_

Laravel 10/11 + Livewire 3.x

Frontend usando Vite (npm run dev / npm run build)

Layout configurado con @vite y @livewireScripts

**_Causas detectadas:_**

Los assets de Livewire no estaban publicados correctamente.

Al no estar publicados, la ruta /livewire/livewire.js no devolvía el JS real sino un error o HTML.

La consola del navegador indicaba que no existía la variable livewire.

**_Solución:_**

1. Publicar los assets de Livewire:
**`php artisan livewire:publish --assets`**
2. Reconstruir los assets de Vite:
`npm run build`
o, para modo desarrollo:
`npm run dev`
